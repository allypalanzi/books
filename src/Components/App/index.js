import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Books from '../Books';
import BookInfo from '../BookInfo';
import Header from '../Header';
import Footer from '../Footer';
import './index.css';

class App extends Component {
  render() {
    return (
      <Router>
        <main className="l-wrapper">
          <Header />
          <Switch>
            <Route exact path={'/'} component={Books} />
            <Route path={'/:id'} component={BookInfo} />
          </Switch>
          <Footer />
        </main>
      </Router>
    );
  }
}

export default App;
