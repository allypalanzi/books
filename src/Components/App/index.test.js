import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './';

jest.mock('../../services');

describe('<App/ >', () => {
  const renderedComponent = renderer.create(<App />);

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
  });

  it('renders correctly', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
