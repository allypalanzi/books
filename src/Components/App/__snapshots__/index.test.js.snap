// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/ > renders correctly 1`] = `
<main
  className="l-wrapper"
>
  <header
    className="c-header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        Science Books 
        <span
          aria-label="Science emojis, scientist, microscope, books"
          role="img"
        >
          👩‍🔬🔬📚
        </span>
      </h1>
    </a>
  </header>
  <section
    className="c-books"
  >
    <a
      className="c-book"
      href="/E7VaaW5hl8kC}"
      onClick={[Function]}
    >
      <img
        alt="Organophosphorus Chemistry cover"
        src="http://books.google.com/books/content?id=E7VaaW5hl8kC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      />
      <h2
        className="c-book__title"
      >
        Organophosphorus Chemistry
      </h2>
    </a>
    <a
      className="c-book"
      href="/PZIdcYCCf2kC}"
      onClick={[Function]}
    >
      <img
        alt="Science and Technology Encyclopedia cover"
        src="http://books.google.com/books/content?id=PZIdcYCCf2kC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      />
      <h2
        className="c-book__title"
      >
        Science and Technology Encyclopedia
      </h2>
    </a>
    <a
      className="c-book"
      href="/9fv1CAAAQBAJ}"
      onClick={[Function]}
    >
      <img
        alt="Pseudomonas syringae and related pathogens cover"
        src="http://books.google.com/books/content?id=9fv1CAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      />
      <h2
        className="c-book__title"
      >
        Pseudomonas syringae and related pathogens
      </h2>
    </a>
    <a
      className="c-book"
      href="/NaB7oAkon9MC}"
      onClick={[Function]}
    >
      <img
        alt="Statistical Physics cover"
        src="http://books.google.com/books/content?id=NaB7oAkon9MC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      />
      <h2
        className="c-book__title"
      >
        Statistical Physics
      </h2>
    </a>
    <a
      className="c-book"
      href="/pSODAgAAQBAJ}"
      onClick={[Function]}
    >
      <img
        alt="Spaces of Social Exclusion cover"
        src="http://books.google.com/books/content?id=pSODAgAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      />
      <h2
        className="c-book__title"
      >
        Spaces of Social Exclusion
      </h2>
    </a>
  </section>
  <footer
    className="c-footer"
  >
    <p>
      ©2018, Thanks Google for your Books API
    </p>
  </footer>
</main>
`;
